src/Conversion/ONNXToTOSA/Math/Softmax.cpp

```diff
@@ -71,16 +82,29 @@ class ONNXSoftmaxLoweringToTOSA : public OpConversionPattern<ONNXSoftmaxOp> {
     // op4 = mul(op1, op3)
     auto op1ExpIn = tosa::CreateOpAndInfer<mlir::tosa::ExpOp>(
         rewriter, op->getLoc(), outputType, adaptor.input());
-    RankedTensorType rsumType =
-        RankedTensorType::get(rsumShape, outputType.getElementType());
-
-    // Keep dims so we don't need to reshape later
-    auto op2ReducesumOp1 =
-        tosa::CreateOpAndInfer<mlir::tosa::ReduceSumOp>(rewriter, op->getLoc(),
-            rsumType, op1ExpIn.getResult(), rewriter.getI64IntegerAttr(axis));
-    auto op3ReciprocalOp2 =
-        tosa::CreateOpAndInfer<mlir::tosa::ReciprocalOp>(rewriter, op->getLoc(),
-            op2ReducesumOp1.getType(), op2ReducesumOp1.getResult());
+    RankedTensorType rsumType = RankedTensorType::get(
+        llvm::SmallVector<int64_t, 4>(inputType.getShape().size(), -1),
+        outputType.getElementType());
+
+    // Get version of softmax. Default is 11.
+    int64_t opVersion = 11;
```

Comment:
My hunch is that there should be different `Softmax` operations here, like we get `Squeeze11` and `Squeeze`. I can accept this for now, but please add a link to the onnx-mlir issue to help people understand.

Before upstreaming though we should wait for feedback on using `gen_onnx_mlir.py` instead of depending on `onnx_opset` being present.

Reply:
I feel the same way. Softmax is the only op that depends on an attribute to determine its version